ulauncher/ui/LayerShell.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import gi
+
+from gi.repository import Gtk
+
+logger = logging.getLogger()
+
+try:
+    gi.require_version('GtkLayerShell', '0.1')
+    # pylint: disable=ungrouped-imports
+    from gi.repository import GtkLayerShell
+except (ValueError, ImportError):
+    logger.warning("Failed to load GtkLayerShell, disabling positioning on wayland!")
+    GtkLayerShell = None
+
+
+class LayerShellOverlay(Gtk.Window):
+    '''
+    Allows for a window to opt in to the wayland layer shell protocol.
+
+    This Disables decorations and displays the window on top of other applications (even if fullscreen).
+    Uses the wlr-layer-shell protocol [1]
+
+    [1]: https://gitlab.freedesktop.org/wlroots/wlr-protocols/-/blob/master/unstable/wlr-layer-shell-unstable-v1.xml
+    '''
+
+    @classmethod
+    def is_supported(cls, *, quiet=False):
+        '''Check if running under a wayland compositor that supports the layer shell extension'''
+        supported = GtkLayerShell is not None and GtkLayerShell.is_supported()
```

Comment:
Nice :+1:  Had exactly something like this in mind for the supported method to work regardless :)